### PR TITLE
axe python2 backend and break out a couple functions, fix tests

### PR DESCRIPTION
**Pull request by cdmistman**
the python impl is a little hard to read/scroll through so i broke a couple functions out of the make* function

i also axed the python2 backend since python2 is dead, long live python3!

### CODE REVIEW

**Review by ryantm — APPROVED** [review]

**Comment by cdmistman** [comment]
oh, *OH* the tests from #144 haven't run until now because i was switching on the wrong name in the test suite 🤦 the test-suite checks for the backend called `python3-poetry` which was actually `python-python3-poetry` until this PR

File: internal/backends/python/python.go [hunk 1]
```diff
@@ -186,10 +214,15 @@ func pythonMakeBackend(name string, python string) api.LanguageBackend {
 			// be a pretty easy fix, though. (Why is this
 			// so complicated??)
 
-			outputB := util.GetCmdOutput([]string{
+			outputB, err := util.GetCmdOutputFallible([]string{
 				"poetry",
 				"config", "settings.virtualenvs.path",
 			})
+			if err != nil {
+				// there's no virtualenv configured, so no package directory
```
Comment (ryantm) [inline internal/backends/python/python.go:222]: Do we need to detect .pythonlibs here somehow? We weren't before and it was working.

File: internal/util/command.go [hunk 2]
```diff
@@ -33,14 +33,22 @@ func RunCmd(cmd []string) {
...
+// GetCmdOutputFallible prints and runs the given command, returning its
+// stdout as a string. Stderr goes to the terminal. GetCmdOutputFallible
+// does not exit the process on error or command failure, but instead
+// returns an error.
+func GetCmdOutputFallible(cmd []string) ([]byte, error) {
+	ProgressMsg(quoteCmd(cmd))
+	command := exec.Command(cmd[0], cmd[1:]...)
+	command.Stderr = os.Stderr
+	return command.Output()
+}
```
Comment (ryantm) [inline internal/util/command.go:45]: nice extraction

**Review by ryantm — APPROVED** [review]

File: internal/backends/python/python.go (hunk 1 above)
Comment (cdmistman) [reply internal/backends/python/python.go:222]: this is mostly to satisfy the test suite. i think something else is going on on replit, but `settings.virtualenvs.path` is broken with `.pythonlibs`
image

i'll take the conversation about how to resolve this for nixmodules offline. i don't think packager3 ever actually calls this function, so i'm not too worried about it being at least slightly broken for the time being

**Pull request merged**